// cdk-dynamo-table-viewer/lib/table-viewer.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.TableViewer = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const path = require("path");
const aws_cdk_lib_1 = require("aws-cdk-lib");
const constructs_1 = require("constructs");
/**
 * Installs an endpoint in your stack that allows users to view the contents of a DynamoDB table through their browser.
 *
 * @stability stable
 */
class TableViewer extends constructs_1.Construct {
    /**
     * @stability stable
     */
    constructor(parent, id, props) {
        super(parent, id);
        const handler = new aws_cdk_lib_1.aws_lambda.Function(this, 'Rendered', {
            code: aws_cdk_lib_1.aws_lambda.Code.fromAsset(path.join(__dirname, '..', 'lambda')),
            runtime: aws_cdk_lib_1.aws_lambda.Runtime.NODEJS_12_X,
            handler: 'index.handler',
            environment: {
                TABLE_NAME: props.table.tableName,
                TITLE: props.title || '',
                SORT_BY: props.sortBy || '',
            },
        });
        props.table.grantReadData(handler);
        const home = new aws_cdk_lib_1.aws_apigateway.LambdaRestApi(this, 'ViewerEndpoint', {
            handler,
            endpointConfiguration: props.endpointType
                ? { types: [props.endpointType] }
                : undefined,
        });
        this.endpoint = home.url;
    }
}
exports.TableViewer = TableViewer;
_a = JSII_RTTI_SYMBOL_1;
TableViewer[_a] = { fqn: "cdk-dynamo-table-viewer.TableViewer", version: "0.2.0" };
